### PR TITLE
Ingest fixes

### PR DESCRIPTION
**Pull request by jameshadfield**
The first 3 commits of #35, as that PR may never be merged.

### CODE REVIEW

File: ingest/rules/upload_from_fauna.smk [hunk 1]
```diff
@@ -1,8 +1,9 @@
 rule download_segment:
     output:
-        sequences = "data/{segment}.fasta",
+        sequences = "data/fauna/{segment}.fasta",
     params:
         fasta_fields = "strain virus accession collection_date region country division location host domestic_status subtype originating_lab submitting_lab authors PMID gisaid_clade h5_clade",
+        output_dir = "data/fauna",
```
Comment (joverlee521) [inline ingest/rules/upload_from_fauna.smk]: _non-blocking_
The Snakemake `params` can access the output, so we can pull out the `output_dir` directly

```suggestion
        output_dir = lambda wildcards, output: Path(output.sequences).parent,
```

**Review by joverlee521 — COMMENTED** [review]
The fixes look good to me! 

One thought that came to me as I was reviewing the fauna namespace change: 
If we are namespacing every independent ingest workflow, I wonder if it would make more sense for the data "source" to be at the top level. 

This would change the structure from 
```
.
└── ingest/
    ├── data/
    │   ├── fauna
    │   └── andersen-lab
    ├── results/
    │   ├── fauna
    │   └── andersen-lab
    └── s3/
        ├── fauna
        └── andersen-lab
```
to 
```
.
└── ingest/
    ├── fauna/
    │   ├── data
    │   ├── results
    │   └── s3
    └── andersen-lab/
        ├── data
        ├── results
        └── s3
```

**Comment by jameshadfield** [comment]
>  If we are namespacing every independent ingest workflow, I wonder if it would make more sense for the data "source" to be at the top level.

I really like the top-level per-source folders idea... let's do this. I have always found the `data` vs `results` distinction within `ingest` to be not quite right, so I'm not thrilled about recreating these within each source directory, but I also don't want to get sidetracked on making progress here.

@joverlee521 do you want to take over this PR and build your NCBI work on top of it?

**Comment by tsibley** [comment]
> If we are namespacing every independent ingest workflow, I wonder if it would make more sense for the data "source" to be at the top level.

+1 for this. It's an approach that's worked well for me in the past when ingesting disparate sources into a single database. Each source has bespoke inputs and processing but emits conventional/standardized outputs which can be used and aggregated by downstream steps.

File: ingest/rules/upload_from_fauna.smk (hunk 1 above)
Comment (joverlee521) [reply ingest/rules/upload_from_fauna.smk]: Updated in https://github.com/nextstrain/avian-flu/pull/36/commits/5e903033875bfcca53a61f6a33436ef5d048ff67

**Comment by joverlee521** [comment]
Tested fauna changes locally with 

```
nextstrain build \
    --envdir ../env.d/seasonal-flu/ \
    ingest upload_all \
        --config "s3_dst=s3://nextstrain-data-private/files/workflows/avian-flu/trial/ingest-fixes" "segments=['ha']"
```
which successfully uploaded the `ha` files to the trial prefix.

Tested andersen-lab changes locally with
```bash
nextstrain build ingest merge_andersen_segment_metadata
```
which successfully completed the ingest. 

---

I'm planning to merge this tomorrow morning. I'll make NCBI ingest changes separately.

**Pull request merged**